addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -752,6 +752,12 @@ MockServer.include({
             });
             if (channel.channel_type === 'channel') {
                 delete res.members;
+            } else {
+                res['seen_partners_info'] = [{
+                    partner_id: this.currentPartnerId,
+                    seen_message_id: channel.seen_message_id,
+                    fetched_message_id: channel.seen_message_id,
+                }];
```

Comment:
Also why do we need this change in the context of the current PR? How is lacking the seen info preventing from opening the DM on receiving a new message?

Reply:
In fact, we need to have a thread with no seen partners info and then it should be updated to something not undefined through the thread convertData method in message_notification_handler. So basically we need the mock server to return this key so the flux goes into the if "seen_partners_info in data"

Reply:
I still don't get why it is needed in this context, for my own understanding. Please explain vocally when you are available.

But this shouldn't prevent from merging since it is there in the python it makes sense to be on the mock server.

Reply:
Ok I can explain vocally if you want. The main idea is that we need to stick to what is really done to reproduce a real situation. There are multiple way to achieve that but what we want is a thread with `partnerSeenInfos` already set on which an insert is called with new values for it, triggering an update of a normally readonly field.

The chosen way here is to have someone talking to current user for the first time, triggering the creation of a channel and a message. Both contains informations about the channel, the one with creation obviously contains the return of the channel_info method. In this process, the message notification is handled first, creating the `partnerSeenInfos` with last fetched message etc THEN the channel notification is handled with the channel info containing `partnerSeenInfos` and trying to update the thread with them. If the field is readonly, then an error is fired (see in `message_notification_handler.js`
![image](https://user-images.githubusercontent.com/48204996/101153411-ae8b0100-3624-11eb-8511-56c11021856c.png))


With this PR, as we removed readonly tag, this won't fail, but the idea was to have a test that will fail as soon as we bring  back the readonly tag.

Reply:
Ok I think I understand now, it's just confusing to introduce a new (and legitimate) test that apparently is completely unrelated to the issue at hand, just because it leads to technically going into some condition of some method.

The issue with this is that if we remove that feature (auto-open of message) we will remove the corresponding test, and we will lose the coverage of some other parts of the code without even noticing.

Reply:
Yes I understand, but I thought that was better than a technical test where I manually trigger 2 following inserts...
Otherwise we could try to find another scenario where two inserts are done, but they would probably done in another feature which could also be disabled/removed afterward.